bluebrain/repo-patches/packages/steps/package.py

```diff
@@ -17,7 +17,7 @@ class Steps(CMakePackage):
     submodules = True
 
     version("develop", branch="master")
-    version("5.0.0a", commit="804e7ef")
+    version("5.0.0a", commit="d75d46f")
```

Comment:
To clarify - If we don’t bump up the version but just change the commit, what is expected here? 🤔

Reply:
The package hash changes along with the `commit` field, so spack should deploy the new version of module 5.0.0a transparently whenever you change the commit referencing this version

Reply:
Ah ok. `new` vs `old` spack behavior then!

Another (important?) aspect of the versioning that I mentioned in our earlier slack conversation:

```console
$ spack-python
Spack version 0.19.1
Python 3.10.8, Linux x86_64
>>> from spack.version import Version
>>> Version('5.0.0') > Version('5.0.0a')
False
```

So in the future, if you release `5.0.0` then `5.0.0a` would be higher! 

Not a solution but workarounds in some other projects IIRC:

```
> >>> Version('5.0.0') > Version('5.0.0.a')                # but similar issue if you try to compare 5.0.0.0
False

```

Reply:
IIRC, it should not. We have _reproducible_ concretization again, and with that, the newest version in the _recipe_ should be chosen.

Reply:
Thank you @pramodk for raising the question, and @matz-e for your clarifications 🥇